wicket-cdi/src/main/java/module-info.java

```diff
@@ -24,6 +24,7 @@
     requires org.apache.wicket.request;
     requires org.apache.wicket.core;
     requires org.slf4j;
+    requires static jakarta.servlet;
```

Comment:
This change is not related to this commit.
I reverted it once and everything was working.
Now it is re-introduced silently again.

Reply:
Please see my latest comment in WICKET-7072 for leaving this change.

Reply:
> Maven Surefire plugin is only for executing the tests via Maven. It does not affect the compilation phase!

I agree, sorry for mixing up compile and surefire plugins.

> In addition: the src/test/java don't have module-info.java, i.e. they are not Java modules!

IMO (esp. after reading https://stackoverflow.com/questions/46613214/java-9-maven-junit-does-test-code-need-module-info-java-of-its-own-and-wher), /src/main/java/module-info.java is used for compiling both /src/main/java and /src/test/java, but compilation of test code seems to be different in Eclipse and Maven, leading to different results (Maven compiles test sources, but Eclipse doesn't do in this case). I think more research is needed here. It would be great if Eclipse would follow Maven in this use case. Meanwhile we could keep requires static jakarta.servlet which would prevent Eclipse developers from manually fixing the build path (via exclude module-info.java, see also my comment in WICKET-7072).

Reply:
> Meanwhile we could keep requires static jakarta.servlet which would prevent Eclipse developers from manually fixing the build path

Fully agree on this! 
I just want to propose an improvement:
- either mention the reason for the change in the commit message
- or even better - use a separate commit for it. Mixing it in the same commit with the wicket-util:test-jar one is not good

Reply:
> Fully agree on this!

Ok, great!

> I just want to propose an improvement:

Thank you for this proposal. I'm unsure how to do this practically:
- revert the change, commit, redo the change with appropriate comment and commit

or

- revert the change, commit, wait until PR is merged into master (with commits squashed), open another PR for WICKET-7072, redo the change with appropriate comment and commit?

Reply:
@martin-g: Looking at some commits in wicket master it looks like that commits of a PR are squashed, and the comments of these commits are merged and used as comment of the squashed commit. Because of this I decided for the first option (reverted the change, commit, redo, added again and commit) and applied it.